lightningd/invoice.c

```diff
@@ -15,18 +15,27 @@
 #include <inttypes.h>
 #include <lightningd/hsm_control.h>
 #include <lightningd/log.h>
+#include <lightningd/options.h>
 #include <sodium/randombytes.h>
 #include <wire/wire_sync.h>
 
 static void json_add_invoice(struct json_result *response,
-			     const struct invoice *inv)
+			     const struct invoice *inv,
+			     bool modern)
 {
 	json_object_start(response, NULL);
 	json_add_string(response, "label", inv->label);
 	json_add_hex(response, "payment_hash", &inv->rhash, sizeof(inv->rhash));
 	if (inv->msatoshi)
 		json_add_u64(response, "msatoshi", *inv->msatoshi);
-	json_add_bool(response, "complete", inv->state == PAID);
+	if (modern)
+		json_add_string(response, "status",
```

Comment:
That's a bit unwieldy, I prefer not to use nested ternaries.

Reply:
I find the below formatting of nested trinaries to be quite wieldy:

    whatever(
        (cond1) ?               expr1 :
        (cond2 && cond3) ?      expr2 - expr3 :
        (cond4 == cond5) ?      expr4 + expr5 :
        /*otherwise*/           default_value);

Reply:
Sure, I'll add a real helper instead.